core/src/main/java/com/datastax/oss/driver/internal/core/protocol/ByteBufCompressor.java

```diff
@@ -25,21 +25,39 @@
 
   @Override
   public ByteBuf compress(ByteBuf uncompressed) {
-    return uncompressed.isDirect() ? compressDirect(uncompressed) : compressHeap(uncompressed);
+    return uncompressed.isDirect()
+        ? compressDirect(uncompressed, true)
+        : compressHeap(uncompressed, true);
   }
 
-  protected abstract ByteBuf compressDirect(ByteBuf input);
+  @Override
+  public ByteBuf compressWithoutLength(ByteBuf uncompressed) {
+    return uncompressed.isDirect()
+        ? compressDirect(uncompressed, false)
+        : compressHeap(uncompressed, false);
+  }
 
-  protected abstract ByteBuf compressHeap(ByteBuf input);
+  protected abstract ByteBuf compressDirect(ByteBuf input, boolean prependWithUncompressedLength);
```

Comment:
nit: switch boolean to not have double negation (false - uncompressed)  - for me, it will be more readable if it will be `preparedWithCompressedLenght`

Reply:
The parameter does not mean "uncompressed vs. compressed", it means "prepend vs. not prepend". If a length gets prepended, it is always the uncompressed one.